testdata/cheats/Fs.t.sol

```diff
@@ -0,0 +1,240 @@
+        cheats.writeLine(path, line1);
+
+        string memory line2 = "second line";
+        cheats.writeLine(path, line2);
+
+        assertEq(cheats.readFile(path), string.concat(line1, "\n", line2, "\n"));
+
+        cheats.removeFile(path);
+
+        cheats.expectRevert(FOUNDRY_WRITE_ERR);
+        cheats.writeLine("/etc/hosts", "malicious stuff");
+    }
+
+    function testCloseFile() public {
+        string memory path = "../testdata/fixtures/File/read.txt";
+
+        assertEq(cheats.readLine(path), "hello readable world");
+        cheats.closeFile(path);
+        assertEq(cheats.readLine(path), "hello readable world");
+    }
+
+    function testRemoveFile() public {
+        string memory path = "../testdata/fixtures/File/remove_file.txt";
+        string memory data = "hello writable world";
+
+        cheats.writeFile(path, data);
+        assertEq(cheats.readLine(path), data);
+
+        cheats.removeFile(path);
+        cheats.writeLine(path, data);
+        assertEq(cheats.readLine(path), data);
+
+        cheats.removeFile(path);
+
+        cheats.expectRevert(FOUNDRY_WRITE_ERR);
+        cheats.removeFile("/etc/hosts");
+    }
+
+    function testWriteLineFoundrytoml() public {
+        string memory root = cheats.projectRoot();
+        string memory foundryToml = string.concat(root, "/", "foundry.toml");
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeLine(foundryToml, "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeLine("foundry.toml", "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeLine("./foundry.toml", "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeLine("./Foundry.toml", "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeLine("./../foundry.toml", "\nffi = true\n");
+    }
+
+    function testWriteFoundrytoml() public {
+        string memory root = cheats.projectRoot();
+        string memory foundryToml = string.concat(root, "/", "foundry.toml");
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeFile(foundryToml, "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeFile("foundry.toml", "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeFile("./foundry.toml", "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeFile("./Foundry.toml", "\nffi = true\n");
+
+        cheats.expectRevert(FOUNDRY_TOML_ACCESS_ERR);
+        cheats.writeFile("./../foundry.toml", "\nffi = true\n");
+    }
+
+    function testReadDir() public {
+        // TODO: Allocation overflow
+        // string memory path = "../testdata/fixtures/Dir";
+
+        // {
+        //     Cheats.DirEntry[] memory entries = cheats.readDir(path);
```

Comment:
Just a single call fails with "allocation overflow", any ideas why?

Reply:
remembered this issue:

https://github.com/foundry-rs/foundry/issues/4409

same error but turned out to be abi decoding related
but perhaps configured memory limit?

Reply:
Thanks, it was an ABI encoding error: 66723e4
Really weird that it throws a "memory overflow" tho

Reply:
yeh, perhaps our mapping of the internal solidity error is not entirely accurate